src/ERC721ContractMetadata.sol

```diff
@@ -176,4 +198,82 @@ contract ERC721ContractMetadata is
     function _baseURI() internal view virtual override returns (string memory) {
         return _tokenBaseURI;
     }
+
+    /**
+     * @notice Returns the contract URI for contract metadata.
+     */
+    function contractURI() external view override returns (string memory) {
+        return _contractURI;
+    }
+
+    /**
+     * @notice Returns the max token supply.
+     */
+    function maxSupply() public view returns (uint256) {
+        return _maxSupply;
+    }
+
+    /**
+     * @notice Returns the provenance hash.
+     *         The provenance hash is used for random reveals, which
+     *         is a hash of the ordered metadata to show it is unmodified
+     *         after mint has started.
+     */
+    function provenanceHash() external view override returns (bytes32) {
+        return _provenanceHash;
+    }
+
+    /**
+     * @notice Returns the address that receives royalties.
+     */
+    function royaltyAddress() external view returns (address) {
+        return _royaltyAddress;
```

Comment:
@Slokh is there a specific function signature we're expecting to get recipient/bps or is that only for multi-recipients?

Reply:
2981 is just a single function that we use, i think these are just helpers ryan added in addition

```
    function royaltyInfo(
        uint256 _tokenId,
        uint256 _salePrice
    ) external view returns (
        address receiver,
        uint256 royaltyAmount
    );
    ```

Reply:
yeah `royaltyAddress` and `royaltyBasisPoints` were just two getters I added for convenience to compliment the setters